webapp/shop/api/ua_contracts/helpers.py

```diff
@@ -252,6 +252,8 @@ def get_user_subscription_statuses(
         ]
 
         statuses["is_trialled"] = True if active_trial else False
+        if not statuses["is_cancelled"]:
```

Comment:
The condition by which a `trial` is auto-renewing is not determined by its cancelled status.

When we call the UAContract API for an account's subscriptions, the subscription payload has a property to tells us if it's auto-renewing or not, we just need to look at that. That property is parsed and converted to the `subscription.is_auto_renewing`.

But that's details. If you replace this `if` with it should fix it:
```
        statuses[
            "is_subscription_auto_renewing"
        ] = is_billing_subscription_auto_renewing(
            subscriptions, subscription_id
        )
```
We do this just below for the `yearly` and `monthly`. It seems that for trials we don't do it, so the default value `False` remains unchanged even if the subscription is auto-renewing.